presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcPreparedStatement.java

```diff
@@ -103,6 +108,106 @@ public void testExecuteQuery()
         }
     }
 
+    @Test
+    public void testGetMetadata()
+            throws Exception
+    {
+        try (Connection connection = createConnection("blackhole", "blackhole")) {
+            try (Statement statement = connection.createStatement()) {
+                statement.execute("CREATE TABLE test_get_metadata (" +
+                        "c_boolean boolean, " +
+                        "c_decimal decimal, " +
+                        "c_decimal_2 decimal(10,3)," +
+                        "c_varchar varchar, " +
+                        "c_varchar_2 varchar(10), " +
+                        "c_row row(x integer, y array(integer)), " +
+                        "c_array array(integer), " +
+                        "c_map map(integer, integer))");
+            }
+
+            try (PreparedStatement statement = connection.prepareStatement(
+                    "SELECT * FROM test_get_metadata")) {
+                ResultSetMetaData metadata = statement.getMetaData();
+                assertEquals(metadata.getColumnCount(), 8);
+                for (int i = 1; i <= metadata.getColumnCount(); i++) {
+                    assertEquals(metadata.getCatalogName(i), "blackhole");
+                    assertEquals(metadata.getSchemaName(i), "blackhole");
+                    assertEquals(metadata.getTableName(i), "test_get_metadata");
+                }
+
+                assertEquals(metadata.getColumnName(1), "c_boolean");
+                assertEquals(metadata.getColumnTypeName(1), "boolean");
+
+                assertEquals(metadata.getColumnName(2), "c_decimal");
+                assertEquals(metadata.getColumnTypeName(2), "decimal(38,0)");
+
+                assertEquals(metadata.getColumnName(3), "c_decimal_2");
+                assertEquals(metadata.getColumnTypeName(3), "decimal(10,3)");
+
+                assertEquals(metadata.getColumnName(4), "c_varchar");
+                assertEquals(metadata.getColumnTypeName(4), "varchar(2147483646)");
+
+                assertEquals(metadata.getColumnName(5), "c_varchar_2");
+                assertEquals(metadata.getColumnTypeName(5), "varchar(10)");
+
+                assertEquals(metadata.getColumnName(6), "c_row");
+                assertEquals(metadata.getColumnTypeName(6), "row()");
```

Comment:
I think this should return `row` (without parenthesis) to be consistent with `array` and `map`. This will need to be fixed in `ClientTypeSignature` to skip parenthesis if the arguments list is empty.